### DISCUSSION TITLE
Binding of: float* listOfFloats

### DISCUSSION THREAD

**Discussion by RichardGe** [wjakob/nanobind#239] [Q&A] [answered]
Hi,
let's say I have a C++ function:

```
void PrintFloats( const float* listOfFloats,  int floatCount )
{
  for(int i=0; i<floatCount ; i++)
      printf("%f\n" , listOfFloats[i] );
}
```

What's the correct way to bind it to python with nanobind ?
Is `m.def("PrintFloats", &PrintFloats);` enough ?
And how can I correctly invoke it on python side ?  For example if I want to call it with 3 numbers:  10, 11, 12, what would be the code ?

sorry, I feel the answer of my question is very obvious but I don't find any answer/example in the doc and I'm not familiar with python.

Thanks.

**Comment by oremanj** [comment, marked as answer]
I would suggest you bind a lambda that accepts `std::vector` and adapts to the interface of PrintFloats. Something like:
```
#include <nanobind/stl/vector.h>
...
m.def("PrintFloats", [](const std::vector<float>& v) {
    PrintFloats(v.data(), v.size());
});
```
Then you would invoke it from Python like `PrintFloats([10, 11, 12])`.

**Comment by RichardGe** [comment]
Thanks, it's working !
